### PR TITLE
[compiler] Make apollo-compiler aware of federation

### PR DESCRIPTION
**Pull request by martinbonnin**
Fixes https://github.com/apollographql/apollo-kotlin/issues/6221

I'm not 100% sold on this solution yet. If other directives appear, we can have the problem again. But maybe it's good that we have this problem? Clients should use introspection results. But sometimes introspection is disabled. And [directives are not exposed in introspection](https://github.com/graphql/graphql-spec/issues/300)...

tldr; this should at least fix the above issue while keeping our future options relatively open.

### CODE REVIEW

**Review by BoD — APPROVED** [review]
Yeah it's a pragmatic tradeoff 👍

**Comment by martinbonnin** [comment]
Closing in favor of https://github.com/apollographql/apollo-kotlin-ffs
